### DISCUSSION TITLE
v32.3.0

### DISCUSSION THREAD

**Discussion by AyanSinhaMahapatra** [nexB/scancode-toolkit#3953] [Announcements]
### Major API/other changes:

* Output Format Version updated to 4.0.0 (major version bump)
*  Dependency attribute rename: is_resolved renamed to is_pinned See https://github.com/aboutcode-org/scancode-toolkit/pull/3888 for more details.
* License Match attribute rename: spdx_license_expression is renamed to license_expression_spdx.

### Changes in Output Data Structure:

* The data structure of the JSON output has changed for: - dependencies at file level package_data, and at top-level. - license matches at file level or unique codebase level license detections Note that the change is a modification to the JSON output, so we have a major version bump 3.2.0 to 4.0.0:
* Dependency attribute is_resolved renamed to is_pinned
* LicenseMatch attribute spdx_license_expression renamed to license_expression_spdx

## What's Changed

* Improve npm workspace processing by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3857
* Fix Package/PackageData creation bugs in purldb by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3710
* Fix python package detection issues #3859 by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3869
* typo in how_to_run_a_scan.rst by @spheex in https://github.com/aboutcode-org/scancode-toolkit/pull/3886
* Fix typo in NuGet package manager reference in README.rst by @Crown0815 in https://github.com/aboutcode-org/scancode-toolkit/pull/3880
* Synchronize Licenses by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3897
* Improve copyright detection by @pombredanne in https://github.com/aboutcode-org/scancode-toolkit/pull/3910
* Improve copyrights detection more by @pombredanne in https://github.com/aboutcode-org/scancode-toolkit/pull/3917
* Update license rules and license detections by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3905
* Update licenses by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3887
* Consolidate Swift package assembly under a single BaseSwiftDatafileHandler by @keshav-space in https://github.com/aboutcode-org/scancode-toolkit/pull/3855
* Improve Copyright Detection by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3929
* Apply miscellaneous license detection updates by @pombredanne in https://github.com/aboutcode-org/scancode-toolkit/pull/3936
* textcode markup: fix SyntaxWarning in regex by @gotmax23 in https://github.com/aboutcode-org/scancode-toolkit/pull/3891
* Update yarn lock parser #3931 by @JonoYang in https://github.com/aboutcode-org/scancode-toolkit/pull/3943
* Detect go binary packages by @pombredanne in https://github.com/aboutcode-org/scancode-toolkit/pull/3894
* Rename dependency is_resolved to is_pinned by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3888
* Rename license match attribute for spdx expression by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3851
* Prepare release v32.3.0 by @AyanSinhaMahapatra in https://github.com/aboutcode-org/scancode-toolkit/pull/3952

## New Contributors
* @spheex made their first contribution in https://github.com/aboutcode-org/scancode-toolkit/pull/3886
* @Crown0815 made their first contribution in https://github.com/aboutcode-org/scancode-toolkit/pull/3880
* @gotmax23 made their first contribution in https://github.com/aboutcode-org/scancode-toolkit/pull/3891

**Full Changelog**: https://github.com/aboutcode-org/scancode-toolkit/compare/v32.2.1...v32.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/aboutcode-org/scancode-toolkit/releases/tag/v32.3.0'>v32.3.0</a>.</em>

**Comment by FaberFlorian** [comment]
Please update the Dockerfile:

```ERROR: Package 'scancode-toolkit' requires a different Python: 3.8.16 not in '>=3.9'```

`FROM --platform=linux/amd64 python:3.8-slim-buster` 
e.g. ->
`FROM --platform=linux/amd64 python:3.9-slim-buster`
but 3.13 is current stable.
